### ISSUE TITLE
Full sync after "Syncing conflict" not working

### ISSUE THREAD

**Issue by hssm** [ankidroid/Anki-Android#3017]
Originally reported on Google Code with ID 2120

```
What steps will reproduce the problem?

1. First we need to have a database on AnkiDroid that differ in such a way from the
database on AnkiWeb that a "Syncing Conflict" occurs.

2. Try to sync from AnkiDroid. Notice "Syncing Conflict" warning message.

3. Choose the option to replace the AnkiDroid database with the database from the server,
AnkiWeb.

4. Now you will see the message "Full sync from server". And you can once again (without
reviewing any cards) press sync. No complaint about "Syncing conflict" this time.

5. *However*, now review one card in AnkiDroid and then press Sync again. The "Syncing
Conflict" warning message pops up again. 


What is the expected output? What do you see instead?
The expected output is that the AnkiDroid database should be fully consistent with
the AnkiWeb version, even after one card review.

Instead the database seemingly differs again after full sync if you review one card.


Does it happen again every time you repeat the steps above? Or did it
happen only one time?

Happens everytime with my current database.


What version of AnkiDroid are you using? (Decks list > menu > About > Look
at the title)
v2.1.3 on both tablet and phone.

On what version of Android? (Home screen > menu > About phone > Android
version)
4.0.4 Galaxy S2 / Tablet



Please provide any additional information below.

* What solved the problem for me was that I reviewed a few cards on desktop Anki. Synced
Anki. Then I successfully synced again on AnkiDroid. The only visible changes in AnkiDroid
database was that the recently reviewed cards were correctly accounted for and removed
from the review-queue.


To summarize: During the whole time I could successfully sync and review cards on desktop
Anki. But even after doing a full sync on AnkiDroid on both my phone and tablet I could
not review any cards without getting a syncing conflict warning message again.
```

Reported by `aron.freidle` on 2014-05-16 13:29:14
